src/Init/Notation.lean

```diff
@@ -552,10 +552,13 @@ except that it doesn't print an empty diagnostic.
 -/
 syntax (name := runMeta) "run_meta " doSeq : command
 
-/-- Element that can be part of a `#guard_msgs` specification. -/
-syntax guardMsgsSpecElt := &"drop"? (&"info" <|> &"warning" <|> &"error" <|> &"all")
+/-- Element of a `#guard_msgs` specification. -/
+syntax guardMsgsSpecElt :=
+  (&"drop"? (&"info" <|> &"warning" <|> &"error" <|> &"all"))
+  <|> (&"whitespace" " := " (&"exact" <|> &"normalized" <|> &"lax"))
+  <|> (&"ordering" " := " (&"exact" <|> &"sorted"))
```

Comment:
I think this is a bit too complex to be a single `syntax`, this kind of thing is really painful to match against because none of the subexpressions are named. I think it would be better to have the three <|> alternatives have their own name, as well as the arguments to `whitespace` and `ordering`. Moreover, I suspect that this is related to the issue you have with the syntax pattern not matching.

Reply:
I ended up not splitting out the `syntax` for the arguments to `whitespace` and `ordering`. I almost did, to be able to split out the parsing functions, but I think there's about the same or even less syntax parsing code this way.

It would be nice to get to the bottom of that syntax match issue though. In principle it should work.

Reply:
Well I'm thinking not just about the code for this function itself but also downstream code that is manipulating these syntaxes, and which may not need to match on every part of it.

Reply:
Do you expect any downstream code to match on these syntaxes? Happy to do it if there's a plausible application.

The only application I can think of is some other command that generates `#guard_msgs` commands with dynamic configurations, but I can't think of what that command would be for. I can only imagine a `macro` with a fixed configuration.

Reply:
Mathport does this kind of thing frequently, for all sorts of commands and syntaxes. Reformatting tools and migration tools will be reading and writing most command syntaxes, and I often had to work around upstream code that overused `("a" <|> "b" <|> "c") "+" ("x" <|> "y" <|> "z")` which is really painful to match.